travharv/templates/execution_report.ttl

```diff
@@ -0,0 +1,59 @@
+{#
+   This file contains the template for the execution report 
+   :param: execution_report 
+   :param: task_reports
+#}
+
+@prefix prov: <http://www.w3.org/ns/prov#>.
+@prefix sh: <http://www.w3.org/ns/shacl#>.
+@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>.
+@prefix xsd: <http://www.w3.org/2001/XMLSchema#>.
+@prefix void: <http://rdfs.org/ns/void#>.
+@prefix schema: <http://schema.org/>.
+@prefix travharv: <http://example.org/travharv/ns/>.
+
+<urn:travharv:{{execution_report.id}}> 
+    a prov:Entity, sh:ValidationReport ;
+
+    prov:generatedAtTime "{{execution_report.last_mod}}" ; 
+    travharv:fromContext "{{execution_report.config_name}}" ;
+    sh:result 
+    {%- for t in task_reports %}{% if not loop.last %}{%- for at in t.assertion_reports %} <urn:travharv:assertionresult-{{at.id}}>,{% endfor %}{% endif %}{% if loop.last %}{%- for at in t.assertion_reports %} <urn:travharv:assertionresult-{{at.id}}>{% if not loop.last %},{% endif %}{% endfor %}{% endif %}{% endfor %};
+.
+
+{# Report of a single travharv tast #}
+{%- for t in task_reports %}
+{%- for at in t.assertion_reports %}
+<urn:travharv:assertionresult-{{at.id}}>
+    a prov:Entity, sh:ValidationResult ; 
+
+    prov:generatedAtTime "{{at.assertion_time}}"^^xsd:dateTime ; 
+
+    sh:focusNode "{{at.subject_uri}}"; {# subject of assertion task #}
+    sh:resultPath "{{at.assertion_path}}"^^xsd:string ; {# property path of assertion task #}
+    sh:resultMessage "{{at.message}}"^^xsd:string ;
```

Comment:
idem

specially important here -- message sounds (more then the other strings in this example) like it could contain ", ' or \n -- all of which will be nicely handled by `| xsd ( "string") `

Reply:
is it weird that I can only use xsd("xsd:string") and others to fix the issue of my templates rendering correctly? @marc-portier @laurianvm